### DISCUSSION TITLE
Styling for a parent or sibling div of a checked radio button/checkbox

### DISCUSSION THREAD

**Discussion by jesse-deboer** [tailwindlabs/tailwindcss#3073] [Ideas]
Sometimes I want to style a parent `div` or a sibling `div` from a checkbox or radio button when it has been checked.

There is [an option](https://tailwindcss.com/docs/hover-focus-and-other-states#checked
) to style an input that has been checked, but nothing for the sibling or parent elements

Maybe there is a way to also style the parent or sibling of a checkbox or radio button with, for example, `group-checked` like `group-hover`?

For the syntax I was thinking about something like the following for sibling:

```
<input type="checkbox" class="sibling" />
<label class="sibling-checked:text-green-500">Checkbox</label>
```

And something like this for the parent/child:

```
<div class="border border-solid border-gray-100 child-checked:border-green-500">
    <input type="checkbox" class="child" />
</div>
```

**Comment by jesse-deboer** [comment]
I just found out this is only possible in the current CSS for the next sibling with a `+`. Maybe it is possible to give this as an option?

**Reply by paulwongx** [reply]
You mean this is only possible via a separate .css file right? I'm facing the same issue, and it doesn't seem possible to use the `+` option in tailwind anywhere except for in a .css file. Please correct me if I'm wrong.

**Comment by stepanjakl** [comment]
This would really be a nice feature to have: there is a case open about this on [Stack Overflow](https://stackoverflow.com/questions/63334626/tailwind-css-is-there-a-way-to-target-next-sibling).

**Comment by VicGUTT** [comment]
Hey !

You'll be glad to hear about the [_peer-* variant_](https://github.com/tailwindlabs/tailwindcss/pull/4556) coming to Tailwind in a couple days ;)

**Reply by simonswiss** [reply]
Came here to say exactly this ☝️🎉

**Reply by VicGUTT** [reply]
Ahah, beat you to it 🚀

**Comment by jesse-deboer** [comment]
Tailwind 2.2 is released and added sibling selector support! 🥳

**Reply by VicGUTT** [reply]
Yep :)

I believe this can now be closed/marked as answered :)

**Reply by simonswiss** [reply]
Only "questions" can be marked as answered in these GitHub discussions – "ideas" remain open _forever_ 👍

**Reply by VicGUTT** [reply]
Oooh, TIL. Also I don't pay enough attention to the "category" attached.
I think it's rather poetic as well, which is pretty much saying "an idea never dies" 👌🏾
